benchpress/benchpress/plugins/hooks/toplev.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+import logging
+
+from benchpress.lib.hook import Hook
+from benchpress.lib import util
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_BACKGROUND_DURATION_SECS = '300'
+DEFAULT_STDOUT_PATH = 'toplev_stdout.log'
+DEFAULT_STDERR_PATH = 'toplev_stderr.log'
+DEFAULT_TOPLEV_OUTPUT_CSV_PATH = 'toplev_output.csv'
+DEFAULT_OPTIONS = [
+    '-l1',  # Level
+    '-I100',
+    '--summary',
+    '-m',  # Extra metrics
+    '--no-desc',
+    '-x,',
+    '-o', DEFAULT_TOPLEV_OUTPUT_CSV_PATH,
+]
```

Comment:
Do we assume that the event map files are downloaded on the system? Maybe add a comment in requirement section to download map files before launching the run with toplev hook.

Reply:
Yes, we  are assuming the user already installed toplev and its dependencies. By default, as long as the host has a network connection to the internet, running toplev for the first time will automatically download the map files off from Intel's website.